client/js/components/addon/AddonWrapper.vue

```diff
@@ -0,0 +1,77 @@
+<license>
+  (c) 2010-present DEMOS E-Partizipation GmbH.
+
+  This file is part of the package demosplan,
+  for more information see the license file.
+
+  All rights reserved
+</license>
+
+<template>
+  <div>
+    <component
+      :is="component"
+      v-bind="getCurrentProperties()"
+    />
+  </div>
+</template>
+
+<script>
+import { checkResponse, dpRpc } from '@demos-europe/demosplan-utils'
+export default {
+  name: 'AddonWrapper',
+
+  props: {
+    hookName: {
+      type: String,
+      required: true,
+      default: ''
+    },
+
+    addonProps: {
+      type: Object,
+      required: false,
+      default: () => {}
+    }
+  },
+
+  data () {
+    return {
+      component: ''
+    }
+  },
+
+  methods: {
+    getCurrentProperties() {
+      return this.addonProps
+    },
+
+    loadComponents(hookName) {
```

Comment:
As `loadComponents` is invoked onMounted, could we also drop the param and access `this.hookName` in line 51?

Reply:
True, we can reduce some lines of code with your suggestion so this was done here https://github.com/demos-europe/demosplan-core/pull/736/commits/283ec1f2756b5b6d46ab5a9cb4035690467ff4f7

Reply:
i actually meant `loadComponents() {` and `dpRpc('addons.assets.load', { hookName: this.hookName })`.

Reply:
I see, sorry I got that wrong. Dropped the function param for directly usage here: https://github.com/demos-europe/demosplan-core/pull/736/commits/f1f753663dcc8909bac7069263ea540ac49add5d